<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>frmMain</class>
 <widget class="QWidget" name="frmMain">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>794</width>
    <height>848</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>OpenGl - Qt -Direct Mode Drawing</string>
  </property>
  <layout class="QGridLayout" name="gridLayout">
   <item row="0" column="1">
    <widget class="GLWidget" name="widget" native="true"/>
   </item>
   <item row="0" column="0">
    <widget class="QFrame" name="frame">
     <property name="sizePolicy">
      <sizepolicy hsizetype="Fixed" vsizetype="Preferred">
       <horstretch>0</horstretch>
       <verstretch>0</verstretch>
      </sizepolicy>
     </property>
     <property name="minimumSize">
      <size>
       <width>180</width>
       <height>0</height>
      </size>
     </property>
     <property name="frameShape">
      <enum>QFrame::StyledPanel</enum>
     </property>
     <property name="frameShadow">
      <enum>QFrame::Raised</enum>
     </property>
     <widget class="QWidget" name="verticalLayoutWidget">
      <property name="geometry">
       <rect>
        <x>10</x>
        <y>10</y>
        <width>164</width>
        <height>152</height>
       </rect>
      </property>
      <layout class="QVBoxLayout" name="verticalLayout_2">
       <property name="spacing">
        <number>0</number>
       </property>
       <property name="sizeConstraint">
        <enum>QLayout::SetNoConstraint</enum>
       </property>
       <item>
        <widget class="QPushButton" name="newCirclePushButton">
         <property name="text">
          <string>New Circle</string>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QPushButton" name="pushButton_3">
         <property name="text">
          <string>New Square</string>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QPushButton" name="pressMePushButton">
         <property name="text">
          <string>Press Me</string>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QPushButton" name="aboutPushButton">
         <property name="text">
          <string>About</string>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QPushButton" name="quitPushButton">
         <property name="text">
          <string>Quit</string>
         </property>
        </widget>
       </item>
      </layout>
     </widget>
     <widget class="QWidget" name="widget_2" native="true">
      <property name="geometry">
       <rect>
        <x>0</x>
        <y>260</y>
        <width>181</width>
        <height>401</height>
       </rect>
      </property>
      <widget class="QDoubleSpinBox" name="set_00">
       <property name="geometry">
        <rect>
         <x>0</x>
         <y>0</y>
         <width>61</width>
         <height>24</height>
        </rect>
       </property>
       <property name="minimum">
        <double>-99.989999999999995</double>
       </property>
       <property name="singleStep">
        <double>0.010000000000000</double>
       </property>
       <property name="value">
        <double>1.000000000000000</double>
       </property>
      </widget>
      <widget class="QDoubleSpinBox" name="set_01">
       <property name="geometry">
        <rect>
         <x>60</x>
         <y>0</y>
         <width>61</width>
         <height>24</height>
        </rect>
       </property>
       <property name="minimum">
        <double>-99.989999999999995</double>
       </property>
       <property name="singleStep">
        <double>0.010000000000000</double>
       </property>
      </widget>
      <widget class="QDoubleSpinBox" name="set_02">
       <property name="geometry">
        <rect>
         <x>120</x>
         <y>0</y>
         <width>61</width>
         <height>24</height>
        </rect>
       </property>
       <property name="minimum">
        <double>-99.989999999999995</double>
       </property>
       <property name="singleStep">
        <double>1.000000000000000</double>
       </property>
      </widget>
      <widget class="QDoubleSpinBox" name="set_10">
       <property name="geometry">
        <rect>
         <x>0</x>
         <y>30</y>
         <width>61</width>
         <height>24</height>
        </rect>
       </property>
       <property name="minimum">
        <double>-99.989999999999995</double>
       </property>
       <property name="singleStep">
        <double>0.010000000000000</double>
       </property>
      </widget>
      <widget class="QDoubleSpinBox" name="set_11">
       <property name="geometry">
        <rect>
         <x>60</x>
         <y>30</y>
         <width>61</width>
         <height>24</height>
        </rect>
       </property>
       <property name="minimum">
        <double>-99.989999999999995</double>
       </property>
       <property name="singleStep">
        <double>0.010000000000000</double>
       </property>
       <property name="value">
        <double>1.000000000000000</double>
       </property>
      </widget>
      <widget class="QDoubleSpinBox" name="set_12">
       <property name="geometry">
        <rect>
         <x>120</x>
         <y>30</y>
         <width>61</width>
         <height>24</height>
        </rect>
       </property>
       <property name="minimum">
        <double>-99.989999999999995</double>
       </property>
       <property name="singleStep">
        <double>1.000000000000000</double>
       </property>
      </widget>
      <widget class="QDoubleSpinBox" name="set_20">
       <property name="geometry">
        <rect>
         <x>0</x>
         <y>60</y>
         <width>61</width>
         <height>24</height>
        </rect>
       </property>
       <property name="minimum">
        <double>-99.989999999999995</double>
       </property>
       <property name="singleStep">
        <double>0.010000000000000</double>
       </property>
      </widget>
      <widget class="QDoubleSpinBox" name="set_21">
       <property name="geometry">
        <rect>
         <x>60</x>
         <y>60</y>
         <width>61</width>
         <height>24</height>
        </rect>
       </property>
       <property name="minimum">
        <double>-99.989999999999995</double>
       </property>
       <property name="singleStep">
        <double>0.010000000000000</double>
       </property>
      </widget>
      <widget class="QDoubleSpinBox" name="set_22">
       <property name="geometry">
        <rect>
         <x>120</x>
         <y>60</y>
         <width>61</width>
         <height>24</height>
        </rect>
       </property>
       <property name="minimum">
        <double>-99.989999999999995</double>
       </property>
       <property name="singleStep">
        <double>1.000000000000000</double>
       </property>
       <property name="value">
        <double>1.000000000000000</double>
       </property>
      </widget>
      <widget class="QPushButton" name="pushButton">
       <property name="geometry">
        <rect>
         <x>0</x>
         <y>110</y>
         <width>181</width>
         <height>32</height>
        </rect>
       </property>
       <property name="text">
        <string>Transfer</string>
       </property>
      </widget>
      <widget class="QPushButton" name="pushButton_2">
       <property name="geometry">
        <rect>
         <x>0</x>
         <y>80</y>
         <width>101</width>
         <height>32</height>
        </rect>
       </property>
       <property name="text">
        <string>Store</string>
       </property>
      </widget>
      <widget class="QPushButton" name="pushButton_4">
       <property name="geometry">
        <rect>
         <x>94</x>
         <y>80</y>
         <width>91</width>
         <height>32</height>
        </rect>
       </property>
       <property name="text">
        <string>Clean</string>
       </property>
      </widget>
      <widget class="QTextEdit" name="textEdit">
       <property name="geometry">
        <rect>
         <x>0</x>
         <y>140</y>
         <width>181</width>
         <height>131</height>
        </rect>
       </property>
       <property name="html">
        <string>&lt;!DOCTYPE HTML PUBLIC &quot;-//W3C//DTD HTML 4.0//EN&quot; &quot;http://www.w3.org/TR/REC-html40/strict.dtd&quot;&gt;
&lt;html&gt;&lt;head&gt;&lt;meta name=&quot;qrichtext&quot; content=&quot;1&quot; /&gt;&lt;style type=&quot;text/css&quot;&gt;
p, li { white-space: pre-wrap; }
&lt;/style&gt;&lt;/head&gt;&lt;body style=&quot; font-family:'.Lucida Grande UI'; font-size:13pt; font-weight:400; font-style:normal;&quot;&gt;
&lt;p style=&quot;-qt-paragraph-type:empty; margin-top:0px; margin-bottom:0px; margin-left:0px; margin-right:0px; -qt-block-indent:0; text-indent:0px;&quot;&gt;&lt;br /&gt;&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
       </property>
      </widget>
      <widget class="QLabel" name="label_2">
       <property name="geometry">
        <rect>
         <x>20</x>
         <y>280</y>
         <width>161</width>
         <height>19</height>
        </rect>
       </property>
       <property name="text">
        <string>The Order of Martix:</string>
       </property>
      </widget>
      <widget class="QLineEdit" name="lineEdit">
       <property name="geometry">
        <rect>
         <x>10</x>
         <y>310</y>
         <width>161</width>
         <height>21</height>
        </rect>
       </property>
       <property name="text">
        <string/>
       </property>
      </widget>
     </widget>
     <widget class="QWidget" name="widget_3" native="true">
      <property name="geometry">
       <rect>
        <x>10</x>
        <y>180</y>
        <width>161</width>
        <height>80</height>
       </rect>
      </property>
      <widget class="QLabel" name="label">
       <property name="geometry">
        <rect>
         <x>10</x>
         <y>0</y>
         <width>41</width>
         <height>16</height>
        </rect>
       </property>
       <property name="text">
        <string>Sides</string>
       </property>
      </widget>
      <widget class="QSpinBox" name="spinBox">
       <property name="geometry">
        <rect>
         <x>50</x>
         <y>0</y>
         <width>101</width>
         <height>24</height>
        </rect>
       </property>
       <property name="minimum">
        <number>3</number>
       </property>
       <property name="maximum">
        <number>1000</number>
       </property>
      </widget>
      <widget class="QPushButton" name="newSquarePushButton">
       <property name="geometry">
        <rect>
         <x>0</x>
         <y>30</y>
         <width>161</width>
         <height>32</height>
        </rect>
       </property>
       <property name="text">
        <string>New Polygon</string>
       </property>
      </widget>
     </widget>
    </widget>
   </item>
  </layout>
 </widget>
 <customwidgets>
  <customwidget>
   <class>GLWidget</class>
   <extends>QWidget</extends>
   <header>glwidget.h</header>
   <container>1</container>
   <slots>
    <slot>newPolygon()</slot>
    <slot>newCircle()</slot>
    <slot>sidesChange(int)</slot>
    <slot>transformShape()</slot>
    <slot>newSquare()</slot>
   </slots>
  </customwidget>
 </customwidgets>
 <resources/>
 <connections>
  <connection>
   <sender>pressMePushButton</sender>
   <signal>clicked()</signal>
   <receiver>frmMain</receiver>
   <slot>pressmebut()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>185</x>
     <y>113</y>
    </hint>
    <hint type="destinationlabel">
     <x>190</x>
     <y>138</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>aboutPushButton</sender>
   <signal>clicked()</signal>
   <receiver>frmMain</receiver>
   <slot>aboutBut()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>106</x>
     <y>141</y>
    </hint>
    <hint type="destinationlabel">
     <x>190</x>
     <y>185</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>quitPushButton</sender>
   <signal>clicked()</signal>
   <receiver>frmMain</receiver>
   <slot>close()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>157</x>
     <y>169</y>
    </hint>
    <hint type="destinationlabel">
     <x>193</x>
     <y>234</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>newSquarePushButton</sender>
   <signal>clicked()</signal>
   <receiver>widget</receiver>
   <slot>newPolygon()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>174</x>
     <y>233</y>
    </hint>
    <hint type="destinationlabel">
     <x>245</x>
     <y>215</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>newCirclePushButton</sender>
   <signal>clicked()</signal>
   <receiver>widget</receiver>
   <slot>newCircle()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>161</x>
     <y>38</y>
    </hint>
    <hint type="destinationlabel">
     <x>229</x>
     <y>55</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>spinBox</sender>
   <signal>valueChanged(int)</signal>
   <receiver>widget</receiver>
   <slot>sidesChange(int)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>160</x>
     <y>205</y>
    </hint>
    <hint type="destinationlabel">
     <x>219</x>
     <y>299</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>pushButton</sender>
   <signal>clicked()</signal>
   <receiver>frmMain</receiver>
   <slot>transferShape()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>97</x>
     <y>382</y>
    </hint>
    <hint type="destinationlabel">
     <x>197</x>
     <y>390</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>pushButton_3</sender>
   <signal>clicked()</signal>
   <receiver>widget</receiver>
   <slot>newSquare()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>163</x>
     <y>59</y>
    </hint>
    <hint type="destinationlabel">
     <x>260</x>
     <y>66</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>pushButton_2</sender>
   <signal>clicked()</signal>
   <receiver>frmMain</receiver>
   <slot>storeMat()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>112</x>
     <y>352</y>
    </hint>
    <hint type="destinationlabel">
     <x>199</x>
     <y>323</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>pushButton_4</sender>
   <signal>clicked()</signal>
   <receiver>frmMain</receiver>
   <slot>cleanStack()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>190</x>
     <y>352</y>
    </hint>
    <hint type="destinationlabel">
     <x>197</x>
     <y>351</y>
    </hint>
   </hints>
  </connection>
 </connections>
 <slots>
  <slot>colourHex()</slot>
  <slot>colourSnake()</slot>
  <slot>showHex(bool)</slot>
  <slot>showSnake(bool)</slot>
  <slot>endit()</slot>
  <slot>widthSlider(int)</slot>
  <slot>panMode(bool)</slot>
  <slot>openImage()</slot>
  <slot>depthSet(int)</slot>
  <slot>clear()</slot>
  <slot>aboutBut()</slot>
  <slot>helpBut()</slot>
  <slot>txt0()</slot>
  <slot>txt0(QString)</slot>
  <slot>txt1()</slot>
  <slot>txt1(QString)</slot>
  <slot>pressmebut()</slot>
  <slot>txt9(QString)</slot>
  <slot>txt99()</slot>
  <slot>transferShape()</slot>
  <slot>storeMat()</slot>
  <slot>cleanStack()</slot>
 </slots>
</ui>
